### PR TITLE
2022-03-03 VT-5510 Use async methods in corresponding sync implementations (where possible)

### PR DESCRIPTION
**Pull request by mrassudishkin**
[VT-5510]
Updated or removed sync methods. They will be use async code now.

[VT-5510]: https://agileharbor.atlassian.net/browse/VT-5510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dk-skuvault — APPROVED** [review]

File: src/BigCommerceAccessTests/Products/ProductsTests.cs [hunk 1]
```diff
@@ -421,15 +232,15 @@ public async Task UpdateProductOptionsAsync_ProductOptionsQuantityUpdated( BigCo
...
 			// Assert
-			var updatedProduct = this.GetProductBySku( service, this.TestProductWithOptionsSku );
+			var updatedProduct = await this.GetProductBySku( service, this.TestProductWithOptionsSku );
 			var updatedProductOption = this.GetProductOptionBySku( updatedProduct, this.TestProductOptionSku );
 			updatedProductOption.Quantity.Should().Be( newOptionQuantity.ToString() );
 		}
 
 		#region Misc
-		private BigCommerceProduct GetProductBySku( IBigCommerceProductsService service, string sku )
+		private async Task< BigCommerceProduct > GetProductBySku( IBigCommerceProductsService service, string sku )
```
Comment (andrey-kozlov-skuvault) [inline src/BigCommerceAccessTests/Products/ProductsTests.cs:241]: Maybe add Async suffix to the name?

File: src/BigCommerceAccessTests/Products/ProductsTests.cs [hunk 2]
```diff
@@ -302,7 +165,7 @@ public void GetProducts_ReturnsProductWhichHaveEnabledTrackingInventoryByProduct
 			// Arrange - Act
 			var config = this.GetConfig( configVersion );
 			var service = this.BigCommerceFactory.CreateProductsService( config );
-			var product = this.GetProductBySku( service, this.TestProductSku );
+			var product = this.GetProductBySku( service, this.TestProductSku ).GetAwaiter().GetResult();
```
Comment (andrey-kozlov-skuvault) [inline src/BigCommerceAccessTests/Products/ProductsTests.cs:168]: Could we make method async and use await?

File: src/BigCommerceAccessTests/Products/ProductsTests.cs [hunk 3]
```diff
@@ -316,36 +179,12 @@ public void GetProducts_ReturnsProductWhichHaveEnabledTrackingInventoryByOption(
 			// Arrange - Act
 			var config = this.GetConfig( configVersion );
 			var service = this.BigCommerceFactory.CreateProductsService( config );
-			var product = this.GetProductBySku( service, this.TestProductWithOptionsSku );
+			var product = this.GetProductBySku( service, this.TestProductWithOptionsSku ).GetAwaiter().GetResult();
```
Comment (andrey-kozlov-skuvault) [inline src/BigCommerceAccessTests/Products/ProductsTests.cs:182]: Could we make method async and use await?

Comment (mrassudishkin) [reply src/BigCommerceAccessTests/Products/ProductsTests.cs:182]: Done

File: src/BigCommerceAccessTests/Products/ProductsTests.cs (hunk 2 above)
Comment (mrassudishkin) [reply src/BigCommerceAccessTests/Products/ProductsTests.cs:168]: Done

File: src/BigCommerceAccessTests/Products/ProductsTests.cs (hunk 1 above)
Comment (mrassudishkin) [reply src/BigCommerceAccessTests/Products/ProductsTests.cs:241]: Done

File: src/BigCommerceAccessTests/Products/ProductsTests.cs [hunk 4]
```diff
@@ -300,12 +163,12 @@ public async Task GetProductsAsync_ReturnsProductsWhichDoesNotBelongToAnyChannel
 		[ TestCase( BigCommerceConfigVersion.NonMsfV2Config ) ]
 		[ TestCase( BigCommerceConfigVersion.NonMsfV3Config ) ]
 		[ TestCase( BigCommerceConfigVersion.MsfV3Config ) ]
-		public void GetProducts_ReturnsProductWhichHaveEnabledTrackingInventoryByProduct( BigCommerceConfigVersion configVersion )
+		public async Task GetProducts_ReturnsProductWhichHaveEnabledTrackingInventoryByProduct( BigCommerceConfigVersion configVersion )
```
Comment (andrey-kozlov-skuvault) [inline src/BigCommerceAccessTests/Products/ProductsTests.cs:166]: Could you also please this and similar change test names to `GetProductsAsync_… `

Comment (mrassudishkin) [reply src/BigCommerceAccessTests/Products/ProductsTests.cs]: Done

**Review by andrey-kozlov-skuvault — APPROVED** [review]

**Pull request merged**